daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -352,19 +353,20 @@ downloadPackage args pid = do
     convPid (LF.PackageId text) = L.PackageId $ TL.fromStrict text
 
 data RemoteDalf = RemoteDalf
-    { remoteDalfName :: String
+    { remoteDalfName :: LF.PackageName
+    , remoteDalfVersion :: Maybe LF.PackageVersion
```

Comment:
This looks like we have to decode all packages even if we only have package ids (either because they are specified in daml.yaml or because we already have a lockfile). Is that required?